src/data_structures/interval_tree/array_backed_interval_tree.rs

```diff
@@ -81,20 +82,22 @@ pub struct ArrayBackedIntervalTree<N: Ord + Clone + Copy, D> {
     indexed: bool,
 }
 
-impl<N: Ord + Clone + Copy, D: Clone> ArrayBackedIntervalTree<N, D> {
-    pub fn new() -> Self {
-        Default::default()
-    }
-
-    pub fn from_iter<I: Iterator<Item = (V, D)>, V>(iter: I) -> Self
-    where
-        V: Into<Interval<N>>,
-    {
+impl<N, D, V> FromIterator<(V, D)> for ArrayBackedIntervalTree<N, D>
+where
+    V: Into<Interval<N>>, N: Ord + Clone + Copy, D: Clone
+{
+    fn from_iter<T: IntoIterator<Item = (V, D)>>(iter: T) -> Self {
         let mut tree = Self::new();
-        iter.for_each(|(interval, data)| tree.insert(interval, data));
+        iter.into_iter().for_each(|(interval, data)| tree.insert(interval, data));
         tree.index();
         tree
     }
```

Comment:
Is this also just a syntax change? This somehow looks like a bigger change, but I guess this just means that the `new()` is implicitly created and the `from_iter` is instead implemented as the `FormIterator` trait? Why does it also need the `into_iter()`, now?
Sorry, just trying to understand what's happening, here.

Reply:
Yes, instead of a function that just happens to be named `from_iter`, actually use the `FromIterator` trait which defines exactly that function. Previously, the function expected an `Iterator`, now it expects an `IntoIterator`, which is why there's the into_iter call